docs/metrics/index.rst

```diff
@@ -0,0 +1,27 @@
+.. _metrics:
+
+==================
+Metrics Collection
+==================
+
+Timeseries metrics my be optionally collected and exported from the server.
+This feature is dissabled by default but can be enabled by passing `--enable-metrics`
+when envoking the server.
+
+The metrics offer insight into how the server is processing messages. Each subsystem publishes
+its own set of metrics. These are timeserises data so they are mostly counters of processed
+elements.
+
+We use the Prometheus metrics format and the metrics are exposed by an HTTP server which by default
+run on for 9215. The implementation is fully complient with Prometheus scraping, so you need only
+point your Prometheus server to the exposed metrics to scrape them.
+
+When using the feature, you need make a directory available which will be used to collect and store
+local metrics from the valious system processes. The server will need rights to write to this
+directory. A directory in ``/tmp`` is fine. Once created you will need to set the
+``prometheus_multiproc_dir`` environment variable which will be read by the server on startup.
+
```

Comment:
Can you please document also each metric and give an example (a little bit enhanced version of the description of this PR would probably work well).

Reply:
@mirceaulinic sure, I can work out most of them, but can you explain what `napalm_logs_publisher_whitelist_blacklist_check_fail` might mean? This is the only one I was a little unclear trying to follow the code.

Reply:
@lampwins: [`error_blacklist`](https://napalm-logs.com/en/latest/publisher/index.html#error-blacklist-raw-unknown) and [`error_whitelist`](https://napalm-logs.com/en/latest/publisher/index.html#error-whitelist) are two new options in 0.4.0. Please check the documentation, and if it doesn't read good, or it's not obvious enough from the first read, please let me know - it means I need to reword it.